src/_pytest/pathlib.py

```diff
@@ -33,7 +33,7 @@ def ensure_reset_dir(path):
     """
     if path.exists():
         rmtree(path, force=True)
-    path.mkdir()
```

Comment:
it absolutely has to fail when doing a ensure_reset

Reply:
The problem as I understand, is a race condition between workers trying to create the same basedir at the same time; worker 1 and 2 (for example) manage to `rmtree` at the same time, but only one of them will succeed when they get to `path.mkdir()`

Reply:
ensure_dir_reset is about completely resetting a folder, if the invariant does not hold, something went baldy wrong, and 2 processes that absolutely should not use the same folder will do so

https://github.com/pytest-dev/pytest/blob/9136b3fda28830c7717d6f4e5a1ac23ee8a1a8f6/src/_pytest/tmpdir.py#L58-L60 - if a exact basetemp is given

it absolutely never must pass if the observed error happens - basetemp invocation itself is not concurrency save ever

Reply:
You are right, my bad, indeed `basetemp` is created by the master process so there's shouldn't be a race condition there.

I'm closing this and the related issue, I will investigate some more on our side as we do pass a basetemp around. 👍 

Thanks for the quick review!